api/apps/api/src/modules/scenarios/scenarios.service.ts

```diff
@@ -88,13 +90,16 @@ export class ScenariosService {
     const validatedMetadata = this.getPayloadWithValidatedMetadata(input);
     const scenario = await this.crudService.create(validatedMetadata, info);
     await this.planningUnitsLinkerService.link(scenario);
+    await this.planningUnitsStatusCalculatorService.calculatedProtectionStatusForPlanningUnitsIn(scenario);
     return scenario;
   }
 
   async update(scenarioId: string, input: UpdateScenarioDTO) {
     await this.assertScenario(scenarioId);
     const validatedMetadata = this.getPayloadWithValidatedMetadata(input);
-    return this.crudService.update(scenarioId, validatedMetadata);
+    const scenario = await this.crudService.update(scenarioId, validatedMetadata);
+    await this.planningUnitsStatusCalculatorService.calculatedProtectionStatusForPlanningUnitsIn(scenario);
```

Comment:
As long as updating (or creating, as above) scenario requires more actions than calling a single crud method (especially, when both are linked together), the relevant change should rather be conducted in standalone service (or inside CRUD service)

Reply:
I'd leave this for a later pass for practical reasons, but how would your suggestion be different from the current implementation that uses a distinct service to set the inclusion status?

Reply:
@hotzevzl the solely purpose of the current service is to be a kind of facade with ACL on top of it (i.e. entry point for module consumer and shared ACL checks), thus being mostly dummy, it has a decent piece of code.

However, the main point of separation is to contain a full logic of create/update in a single place - to both having common sense what creating means together with what are the side effects and required actions.